plugins/networktables/src/main/java/edu/wpi/first/shuffleboard/plugin/networktables/sources/CompositeNetworkTableSource.java

```diff
@@ -35,20 +37,22 @@ public CompositeNetworkTableSource(String tableName, ComplexDataType<D> dataType
     super(tableName, dataType);
     this.dataType = dataType;
     String path = NetworkTableUtils.normalizeKey(tableName, false);
-    ITable table = NetworkTable.getTable(path);
+    NetworkTable table = NetworkTableInstance.getDefault().getTable(path);
     setData(dataType.getDefaultValue());
 
-    setTableListener((key, value, flags) -> {
-      boolean delete = NetworkTableUtils.isDelete(flags);
-      String relativeKey = NetworkTableUtils.normalizeKey(key.substring(path.length() + 1), false);
-      if (delete) {
-        backingMap.remove(relativeKey);
-      } else {
-        backingMap.put(relativeKey, value);
-      }
-      setActive(Objects.equals(NetworkTableUtils.dataTypeForEntry(fullTableKey), dataType));
-      setData(dataType.fromMap(backingMap));
-    });
+    table.addEntryListener(
```

Comment:
This shouldn't have been changed from calling `setTableListener`. This callback won't run on the JavaFX application thread, which causes issues.